src/Message/Broker/Redis.php

```diff
@@ -47,22 +47,29 @@ public function hasMessage(): bool
             throw $throwable;
         }
 
-        return $publishChannelLength + $subscriptionChannelLength > 0 ? true : false;
+        return ($publishChannelLength + $subscriptionChannelLength > 0);
     }
 
-    public function getNextMessage(): string
+    public function attemptGetNextMessage(): ?string
     {
         try {
-            $message = $this->_getRedisClient()->lPop($this->_getSubscriptionChannelName());
-            if ($message === false) {
-                $message = $this->_getRedisClient()->rPop($this->_getPublishChannelName());
+            $subscriptionMessage = $this->_getRedisClient()->lPop($this->_getSubscriptionChannelName());
+
+            if ($subscriptionMessage !== false) {
+                return $subscriptionMessage;
+            }
+
+            $publicationMessage = $this->_getRedisClient()->rPop($this->_getPublishChannelName());
+
+            if ($publicationMessage !== false) {
+                return $publicationMessage;
             }
+
+            return null;
```

Comment:
Good job unpacking this flow. Now that's it's more clear, it's easier to see we have two messages sources we care about from either `subscription` and `publish` channels. 

Between these two we want to process messages from the `subscription` first. From that channel we want the oldest (left most) message. If nothing is there, then we grab the newest (right most) message from the publication channel. 

Can you explain why `lPop()` and `rPop()` is used in each case?

Reply:
Just following the existing state of the world, I don't fully understand it

Reply:
@alexberryman - ordering wrt `brpoplpush` on line 18